### PR TITLE
Backend for Customer Success Platform

### PR DESCRIPTION
**Pull request by Priyanshu-792**
To perform all the cruds successfully.

### CODE REVIEW

File: Promact.CustomerSuccess.Platform/Controllers/EmailController.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+using System.Text;
+using Org.BouncyCastle.Tls;
+using System.Text.RegularExpressions;
+
+
+namespace Promact.CustomerSuccess.Platform.Entities
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class EmailController : ControllerBase
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Controllers/EmailController.cs:14]: In abp.io we don't create controllers. Review this please https://docs.abp.io/en/abp/latest/Tutorials/Part-1?UI=MVC&DB=EF#bookappservice

File: Promact.CustomerSuccess.Platform/Controllers/EmailController.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    {
+        private const string EmailHost = "smtp.gmail.com";
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Controllers/EmailController.cs:16]: Don't put sensitive details in code

File: Promact.CustomerSuccess.Platform/Controllers/EmailController.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        private const int EmailPort = 587;
+        private const string EmailUsername = "priyanshusharma72002@gmail.com";
+        private const string EmailPassword = "dcovmvshssnpwmjo";
+
+        [HttpPost("send")]
+        public IActionResult SendEmails([FromBody] List<EmailDto> emails)
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Controllers/EmailController.cs:22]: Lack of async await

File: Promact.CustomerSuccess.Platform/Controllers/EmailController.cs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            }
+        }
+    }
+
+
+
+
+}
+
+    public class EmailDto
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Controllers/EmailController.cs:93]: Don't put classes randomly

File: Promact.CustomerSuccess.Platform/Entities/ApprovedTeam.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
+﻿using System.ComponentModel.DataAnnotations.Schema;
+using Volo.Abp.Domain.Entities.Auditing;
+
+namespace Promact.CustomerSuccess.Platform.Entities
+{
+    public class ApprovedTeam : AuditedAggregateRootWithUser<Guid, ApplicationUser>
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Entities/ApprovedTeam.cs:6]: What is the purpose of this table?

File: Promact.CustomerSuccess.Platform/Entities/DetailedTimeline.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel.DataAnnotations.Schema;
+using Volo.Abp.Domain.Entities;
+
+namespace Promact.CustomerSuccess.Platform.Entities
+{
+    public class DetailedTimeline : Entity<Guid>
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Entities/DetailedTimeline.cs:6]: What is the purpose of this table?

File: Promact.CustomerSuccess.Platform/Entities/EscalationMatrix.cs [hunk 7]
```diff
@@ -6,9 +6,11 @@ namespace Promact.CustomerSuccess.Platform.Entities
     public class EscalationMatrix : AuditedAggregateRootWithUser<Guid, ApplicationUser>
     {        
         public EscalationMatrixLevels Level { get; set; }
-        public EscalationType EscalationType { get; set; }        
         [ForeignKey("Project")]
         public required Guid ProjectId { get; set; }
+        public EscalationType EscalationType { get; set; }
+        public string Name { get; set; }
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Entities/EscalationMatrix.cs:12]: What do you mean by Name here?

File: Promact.CustomerSuccess.Platform/Entities/PhaseMilestone.cs [hunk 8]
```diff
@@ -7,14 +7,14 @@ public class PhaseMilestone : AuditedAggregateRootWithUser<Guid, ApplicationUser
...
         public required string Title { get; set; }
         public DateTime StartDate { get; set; }
         public DateTime EndDate { get; set; }
-        public required string Description { get; set; }
-        public required string Comments { get; set; }
+        public DateTime ApprovalDate { get; set; }
         public MilestoneOrPhaseStatus Status { get; set; }
-        public virtual Project? Project { get; set; }
-        public virtual ICollection<Sprint>? Sprints { get; set; }
+        public DateTime RevisedCompletionDate { get; set; }
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Entities/PhaseMilestone.cs:16]: What is the purpose of this field?

File: Promact.CustomerSuccess.Platform/Entities/ProjectDescription.cs [hunk 9]
```diff
@@ -0,0 +1,18 @@
+﻿
+using System.ComponentModel.DataAnnotations.Schema;
+using Volo.Abp.Domain.Entities;
+
+namespace Promact.CustomerSuccess.Platform.Entities
+{
+    public class ProjectDescription : Entity<Guid>
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Entities/ProjectDescription.cs:7]: Why is this a separate table for description? What is the benefit of keeping it in different tables?

File: Promact.CustomerSuccess.Platform/Entities/ProjectResources.cs [hunk 10]
```diff
@@ -8,16 +8,14 @@ public class ProjectResources : AuditedAggregateRootWithUser<Guid, ApplicationUs
         [ForeignKey("Project")]
         public Guid ProjectId { get; set; }
         public virtual Project? Project { get; set; }
-        [ForeignKey("Resource")]
-        public Guid ResourceId { get; set; }
         public virtual Guid Resource { get; set; }
+        public string ResourceName { get; set; }
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Entities/ProjectResources.cs:12]: What is the purpose of this?

File: Promact.CustomerSuccess.Platform/Entities/ProjectUpdates.cs [hunk 11]
```diff
@@ -0,0 +1,15 @@
+﻿using System.ComponentModel.DataAnnotations.Schema;
+using Volo.Abp.Domain.Entities.Auditing;
+
+namespace Promact.CustomerSuccess.Platform.Entities
+{
+    public class ProjectUpdates : AuditedAggregateRootWithUser<Guid, ApplicationUser>
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Entities/ProjectUpdates.cs:6]: What is the purpose of this table?

File: Promact.CustomerSuccess.Platform/Entities/RiskProfile.cs [hunk 12]
```diff
@@ -7,11 +7,15 @@ public class RiskProfile : AuditedAggregateRootWithUser<Guid, ApplicationUser>
...
         public Guid ProjectId { get; set; }
+        public virtual Project? Project { get; set; }
         public RiskType RiskType { get; set; }
         public RiskSeverity Severity { get; set; }
-        public RiskImpact Impact { get; set; }        
-        public virtual ICollection<RemediationStep>? RemediationSteps { get; set; }
-        public virtual Project? Project { get; set; }
+        public RiskImpact Impact { get; set; }
+        public string? Description { get; set; }
+        public string? RemedialSteps { get; set; }
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Entities/RiskProfile.cs:15]: Why RemedialSteps have been converted to a string field from a different table?

File: Promact.CustomerSuccess.Platform/Entities/RiskProfile.cs [hunk 13]
```diff
@@ -7,11 +7,15 @@ public class RiskProfile : AuditedAggregateRootWithUser<Guid, ApplicationUser>
...
         [ForeignKey("Project")]
         public Guid ProjectId { get; set; }
+        public virtual Project? Project { get; set; }
         public RiskType RiskType { get; set; }
         public RiskSeverity Severity { get; set; }
-        public RiskImpact Impact { get; set; }        
-        public virtual ICollection<RemediationStep>? RemediationSteps { get; set; }
-        public virtual Project? Project { get; set; }
+        public RiskImpact Impact { get; set; }
+        public string? Description { get; set; }
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Entities/RiskProfile.cs:14]: What is the purpose of Description?

File: Promact.CustomerSuccess.Platform/Entities/Scope.cs [hunk 14]
```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel.DataAnnotations.Schema;
+using Volo.Abp.Domain.Entities;
+
+namespace Promact.CustomerSuccess.Platform.Entities
+{
+    public class Scope : Entity<Guid>
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Entities/Scope.cs:6]: I cannot understand creating tables with just a 1-to-1 relation between entities. correct me if I am wrong here

File: Promact.CustomerSuccess.Platform/Entities/Sprint.cs
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Entities/Sprint.cs:1]: Now why did we remove link with Phased Milestone

File: Promact.CustomerSuccess.Platform/Entities/StakeHolders.cs [hunk 15]
```diff
@@ -0,0 +1,15 @@
+﻿using System.ComponentModel.DataAnnotations.Schema;
+using Volo.Abp.Domain.Entities;
+
+namespace Promact.CustomerSuccess.Platform.Entities
+{
+    public class StakeHolders : Entity<Guid>
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Entities/StakeHolders.cs:6]: Who are the stakeholders? What is the purpose of this table?

File: Promact.CustomerSuccess.Platform/Migrations/20240313095405_projectBudget.cs
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Migrations/20240313095405_projectBudget.cs:1]: Why do we have a blank migration?

File: Promact.CustomerSuccess.Platform/Migrations/20240313164936_document.cs
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Migrations/20240313164936_document.cs:1]: Why blank?

File: Promact.CustomerSuccess.Platform/Migrations/20240313213528_Risk Profilings.cs
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Migrations/20240313213528_Risk Profilings.cs:1]: Again?

File: Promact.CustomerSuccess.Platform/Migrations/20240314061342_PhaseMilestones.cs
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Migrations/20240314061342_PhaseMilestones.cs:1]: Again?

File: Promact.CustomerSuccess.Platform/Services/ClientFeedbackService.cs [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+using Volo.Abp.Domain.Repositories;
+
+namespace Promact.CustomerSuccess.Platform.Services
+{
+    public class ClientFeedbackService :
+   CrudAppService<
+            ClientFeedback,
+            ClientFeedbackDto,
+
+            Guid,
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Services/ClientFeedbackService.cs:14]: Fix the formatting

File: Promact.CustomerSuccess.Platform/Services/Dtos/CreateApprovedTeamDto.cs [hunk 17]
```diff
@@ -0,0 +1,17 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace Promact.CustomerSuccess.Platform.Services.Dtos
+{
+    public class CreateApprovedTeamDto
```
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Services/Dtos/CreateApprovedTeamDto.cs:5]: Why have you not implemented IEntityDto here and other DTO classes?

File: Promact.CustomerSuccess.Platform/Services/IApprovedTeamService.cs
Comment (chintans) [inline Promact.CustomerSuccess.Platform/Services/IApprovedTeamService.cs:1]: Why empty?

**Review by chintans — CHANGES_REQUESTED** [review]
naming conventions in generating migrations are wrong, we use pascle casing and no spaces in the names.

File: Promact.CustomerSuccess.Platform/Controllers/EmailController.cs (hunk 1 above)
Comment (Priyanshu-792) [reply Promact.CustomerSuccess.Platform/Controllers/EmailController.cs:14]: Thank you sir for providing the reference.

File: Promact.CustomerSuccess.Platform/Controllers/EmailController.cs (hunk 2 above)
Comment (Priyanshu-792) [reply Promact.CustomerSuccess.Platform/Controllers/EmailController.cs:16]: > Don't put sensitive details in code

Ok Sir thankyou for your feedback i will try to resolve this.

File: Promact.CustomerSuccess.Platform/Controllers/EmailController.cs (hunk 3 above)
Comment (Priyanshu-792) [reply Promact.CustomerSuccess.Platform/Controllers/EmailController.cs:22]: I will implement it.

File: Promact.CustomerSuccess.Platform/Controllers/EmailController.cs (hunk 4 above)
Comment (Priyanshu-792) [reply Promact.CustomerSuccess.Platform/Controllers/EmailController.cs:93]: ok sir i will try to put it in the proper dto folder.

File: Promact.CustomerSuccess.Platform/Entities/ApprovedTeam.cs (hunk 5 above)
Comment (Priyanshu-792) [reply Promact.CustomerSuccess.Platform/Entities/ApprovedTeam.cs:6]: It was present in our guidelines and requirements of the project, to store approved team for every project.

File: Promact.CustomerSuccess.Platform/Migrations/20240313213528_Risk Profilings.cs
Comment (Priyanshu-792) [reply Promact.CustomerSuccess.Platform/Migrations/20240313213528_Risk Profilings.cs:1]: Actually sir , initially i was not aware when to create migration , so any changes i was doing in backend code i was creating migration thinking that it is a convention to do it. But anyways, i will definitely take care of it not to provide any empty migration further.

File: Promact.CustomerSuccess.Platform/Migrations/20240314061342_PhaseMilestones.cs
Comment (Priyanshu-792) [reply Promact.CustomerSuccess.Platform/Migrations/20240314061342_PhaseMilestones.cs:1]: I will rectify it sir.

File: Promact.CustomerSuccess.Platform/Services/ClientFeedbackService.cs (hunk 16 above)
Comment (Priyanshu-792) [reply Promact.CustomerSuccess.Platform/Services/ClientFeedbackService.cs:14]: ok

**Pull request closed without merging**